wgpu/src/triangle.rs

```diff
@@ -261,65 +272,85 @@ impl Pipeline {
                 * Transformation::translate(mesh.origin.x, mesh.origin.y))
             .into();
 
-            let vertex_buffer = device.create_buffer_with_data(
-                bytemuck::cast_slice(&mesh.buffers.vertices),
-                wgpu::BufferUsage::COPY_SRC,
-            );
+            let vertices = bytemuck::cast_slice(&mesh.buffers.vertices);
+            let indices = bytemuck::cast_slice(&mesh.buffers.indices);
 
-            let index_buffer = device.create_buffer_with_data(
-                mesh.buffers.indices.as_bytes(),
-                wgpu::BufferUsage::COPY_SRC,
-            );
+            if let Some(vertices_size) =
```

Comment:
nit: combine two pattern matches into one?

Reply:
Done in 07880c392c865d700cbaff7601c1fffd3354dd4f.